### DISCUSSION TITLE
It is possible to use Tanstack Router with Atlassian Forge implementation?

### DISCUSSION THREAD

**Discussion by aleksolutions** [TanStack/router#1836] [Q&A]
Hi guys,

I'm trying to use Tantstack Router inside an Atlassian Cloud addon with their framework called Forge. The problem I'm facing is that I don't know if their implementation of "history" [createHistory](https://developer.atlassian.com/platform/forge/apis-reference/ui-api-bridge/view/#createhistory) would be possible to somehow translate it to put it inside the tanstack router object.

The function signature is in the previous link.

So, would it be feasible to integrate it with this object that they provide from their api?

Thanks!

**Comment by SeanCassiere** [comment]
I'm not quite sure if that'd work, since TSR uses our own implementation of [history](https://github.com/TanStack/router/tree/main/packages/history).
